### ISSUE TITLE
Allow to move branch around

### ISSUE THREAD

**Issue by correnson** [ForkIssues/Tracker#190]
The UI lacks the ability of moving a branch (not the current one) to another commit — or maybe I missed it?

The excellent (but outdated) [gitx](http://gitx.frim.nl/index.html) provides Drag&Drop for this feature.
Alternatively, on the contextual menu on branch « A », when branch « Main » is checked out, it would be nice to have the actions « Reset 'Main' to 'A' » and « Reset 'A' to 'Main' »

**Comment by DanPristupov** [comment]
I didn't know about this ability. Just tried, it works nicely.

While it looks cool it can also be dangerous, I think. If branch is a leaf, dragging it to another commit will cause loosing its commits.

**Comment by correnson** [comment]
Yes indeed. A possible protection would be to only allow this action is there is another branch on the leaf, or if it is not leaf.

Another useful action in contextual menu is to add a new branch on an specific commit. It combines very well with the proposals 😺

**Comment by passsy** [comment]
I maintain `stable` and `beta` branches that I move around freely between commits. They often sit on hot-fix branches and when moving their history is usually not compatible due to cherry-picks.

Moving the branch around currently requires:

```bash
$ git branch -d beta
Deleted branch beta (was 5321b3f).
$ git branch --force beta 8970cf90d597181aab1ce0254ec1c39c2f2a048f
```

Drag 'n' drop might be a bit too much for this rare use case, but a context menu item on the branch `Move to commit <sha1>` would do the job nicely
